hopsworks-admin/src/main/java/io/hops/hopsworks/admin/ndb/NdbBackupBean.java

```diff
@@ -100,15 +106,22 @@ public String restore(Integer backupId) {
     String prog = settings.getHopsworksDomainDir() + "/bin/ndb_backup.sh";
     int exitValue;
 
-    String[] command = {"/usr/bin/sudo", prog, "restore", backupId.toString()};
-    ProcessBuilder pb = new ProcessBuilder(command);
+    ProcessDescriptor processDescriptor = new ProcessDescriptor.Builder()
+        .addCommand("/usr/bin/sudo")
+        .addCommand(prog)
+        .addCommand("restore")
+        .addCommand(backupId.toString())
+        .setWaitTimeout(2L, TimeUnit.HOURS)
```

Comment:
Do we know 2 hours is enough, how long does it take now? Might be hard to accurately put an upper limit on this.

Reply:
It is hard to put an upper limit in this case. 2 hours it might be OK. But in any case we should either modify the back-up script not to wait for NDB backup to complete or keep using the backup mechanism that is already in place.